apps/site/pages/docs/api-extensions/extending-queries-using-fragments.mdx

```diff
@@ -0,0 +1,241 @@
+---
+title: ' Extending queries using fragments'
+sidebar_label: 'API Extensions'
+---
+
+import { Callout, Tabs, Tab } from 'nextra-theme-docs'
+import { ExtendableQueryTable } from './../../../components/ExtendableQueryTable'
+
+<header>
+
+# Extending queries using fragments
+
+</header>
+
+After [defining the new fields exposed by the FastStore API Extension](/docs/api-extension/extending-api-schema), the next step is to specify where these fields will be utilized within GraphQL fragments.
+
+[Fragments](https://graphql.org/learn/queries/#fragments) in GraphQL are reusable units that enhance query functionality. FastStore associates these fragments with the existing queries used on its pages to incorporate the newly exposed fields.
+
+<Callout type="info" emoji="ℹ️">
+  To access the list of queries and their corresponding fragments, refer to the [extendable queries](/docs/api-extension/extending-queries-using-fragments#extendable-queries) section.
+</Callout>
+
+follow the steps below to add custom fields to an [existing query](/docs/api-extensions#extendable-queries). We will use the `ServerProductPage` query as an example to illustrate how to extend.
+
+---
+## Before you start
+
+### Avoid over-fetching data on pages
+ 
+Even though you can add information to the FastStore API schema, you must be careful not to over-fetch data on your pages. See the [best practices for fetching data on your storefront](https://v1.faststore.dev/how-to-guides/faststore-api/fetching-api-data#best-practices-for-fetching-data).
+
+---
+
+## Step by step
+
+### Step 1: Create a new file
+
+1. In the `src` folder of your store code, create the `fragments` folder.
+
+  ```bash copy
+  mkdir src fragments
+  ```
```

Comment:
We are already in the src folder. So we just need to create the fragments folder.
```suggestion
1. In the `src` folder of your store code, create the `fragments` folder.

  ```bash copy
  mkdir fragments
  ```
```